frontend/src/libs/components/input/input.tsx

```diff
@@ -41,32 +45,73 @@ const Input = <T extends FieldValues>({
   const hasError = Boolean(error);
   const hasValue = Boolean(field.value);
   const hasLabel = Boolean(label);
-  const inputStyles = [
-    styles.input,
-    hasValue && styles.filled,
-    isDisabled && styles.disabled,
-    hasError && styles.error,
-  ];
+
+  const controlledInputStyles = useMemo(
+    () => [
+      hasValue && styles.filled,
+      isDisabled && styles.disabled,
+      hasError && styles.error,
+    ],
+    [hasError, hasValue, isDisabled],
+  );
+
+  const defaultInputStyles = useMemo(
+    () => [styles.input, styles.inputPadding, ...controlledInputStyles],
+    [controlledInputStyles],
+  );
+
```

Comment:
do u need to wrap it in useMemo?

Reply:
yes, eslint suggest it: The 'controlledInputStyles' array makes the dependencies of useCallback Hook (renderPhoneInput) change on every render. To fix this, wrap the initialization of 'controlledInputStyles' in its own useMemo()